task_management/app/models/task.rb

```diff
@@ -9,6 +9,22 @@ class Task < ApplicationRecord
   validates :name, presence: true, length: { maximum: 50 }
   validates :deadline, presence: true
 
-  enum priority: { low: 0, normal: 1, high: 2 }
-  enum status: { todo: 0, in_progress: 1, done: 2 }
+  enum priority: { low: 1, normal: 2, high: 3 }
+  enum status: { todo: 1, in_progress: 2, done: 3 }
+
+  scope :get_status, (lambda do |status|
+    where(status: status) if statuses.keys.include?(status)
+  end)
```

Comment:
すごくスッキリして読みやすくなったと思いますー！
ただ、Railsエンジニアだと以下のように書く人が多いかも？個人的な意見ですが不要な情報が少なくて読みやすい気がしますー
```suggestion
  scope :get_status, -> (status) {
    where(status: status) if statuses.keys.include?(status)
  }
```
少なくとも以下のように `()` は消してほしいかも？あとはお好みでー
```ruby
  scope :get_status, lambda do |status|
    where(status: status) if statuses.keys.include?(status)
  end
```

Reply:
ありがとうございます！

本件について質問があります。。。

１．↑でいただいたサンプルのようにラムダ式をアローで複数行書くとrubocopにて以下の警告が発生します
`RuboCop: Do not use spaces between `->` and opening brace in lambda literals [Layout/SpaceInLambdaLiteral]`

こちら、ラクマでは許容しているのでしょうか？

２．「１」を許容していない場合、書き方は以下の２通りになるかと思うのですが、認識相違ないですか？
パターン①  ※`()`は文法的に消せなさそう・・
```
scope :get_status, (lambda do |status|
    where(status: status) if statuses.keys.include?(status)
  end)
```

パターン②
```
gs = lambda do |status|
  where(status: status) if statuses.keys.include?(status)
  end
scope :get_status, gs
```

■参考資料
[Rubocop reports an error on multiline lambdas in scopes](https://github.com/rubocop-hq/rubocop/issues/1520)

Reply:
おお、ほんとですね。すみません、確認不足でした！
以下のどちらかでなら大丈夫だと思いますー

```ruby
  scope :get_status, ->(status) { where(status: status) if statuses.keys.include?(status) }

  scope :get_status, lambda do |status|
    where(status: status) if statuses.keys.include?(status)
  end
```

Reply:
すみません、また連絡不足がありました・・

↑１つ目のサンプルですと、rubocopにて以下の警告が発生します
`RuboCop: Line is too long. [91/80] [Metrics/LineLength]`

２つ目のサンプルですと、scopeの文法的に`()`は消せません・・

となると、↓の「２．①②」のいずれかの書き方になると考えております！
https://github.com/Fablic/training/pull/909#discussion_r565812664

Reply:
あ、ほんとだ、 `()` ないとダメですね、、、

Reply:
> ↑１つ目のサンプルですと、rubocopにて以下の警告が発生します
RuboCop: Line is too long. [91/80] [Metrics/LineLength]

あれ、これ現在は160文字に緩くなってそうだけどなんでだろ

Reply:
いや、120文字になってるかな？

Reply:
私の環境に「.base_rubocop.yml」を反映できていませんでした・・

以下ファイルにて最大文字数が160文字になっていました。
> Metrics/LineLength:
  Max: 160

https://github.com/Fablic/fablicop/blob/master/config/.base_rubocop.yml

Reply:
scopeを可能なものは１行にしました！